ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -714,18 +729,21 @@ void checkAndMaybeComplete() {
 
     /**
      * Handle the body of the response: Deserialize and add to the list of chunk buffers.
+     * @param responseInfo the response received for a request sent out on behalf of this chunk.
      * @param payload the body of the response.
      * @param messageMetadata the {@link MessageMetadata} associated with the message.
      * @param messageInfo the {@link MessageInfo} associated with the message.
      * @throws IOException if there is an IOException while deserializing the body.
      * @throws MessageFormatException if there is a MessageFormatException while deserializing the body.
      */
-    void handleBody(InputStream payload, MessageMetadata messageMetadata, MessageInfo messageInfo)
-        throws IOException, MessageFormatException {
+    void handleBody(ResponseInfo responseInfo, InputStream payload, MessageMetadata messageMetadata,
+        MessageInfo messageInfo) throws IOException, MessageFormatException {
       if (!successfullyDeserialized) {
         BlobData blobData = MessageFormatRecord.deserializeBlob(payload);
         ByteBuffer encryptionKey = messageMetadata == null ? null : messageMetadata.getEncryptionKey();
         ByteBuffer chunkBuffer = blobData.getStream().getByteBuffer();
+        responseInfo.retain();
```

Comment:
It's one. When a buffer is created, the refcount will be initialized to one. And this method would increase the refcount by one.

Reply:
ah right, in `OperationController` that first ref will be decremented and this will increment the ref count when handing off the buffer to a different context